ath79: add support for TP-Link TL-WR841HP v3

andyteq opened the pull request:
Specifications:
- QCA9533 SoC, 8 MiB nor flash, 64 MiB DDR2 RAM
- 4x Ethernet LAN 10/100, 1x Ethernet WAN 10/100
- 1x LAN, WAN, Wifi, PWR, WPS, RE Leds
- Reset, Wifi on/off, WPS, RE buttons
- Serial UART at J4 onboard: 3.3v GND RX TX, 1152008N1

Installation:

Upload openwrt-ath79-generic-tplink_tl-wr841hp-v3-squashfs-factory.bin
from stock firmware webgui

Revert back to stock firmware instructions:

- set your PC to static IP address 192.168.0.66 netmask 255.255.255.0
- download stock firmware from Tp-link website
- put it in the root directory of tftp server
- rename it to wr841hpv3_tp_recovery.bin
- power on while pressing reset button until any LED is lighting up
- wait for the router to reboot. done

This port has been tested by many people, see here
https://forum.openwrt.org/t/support-for-tp-link-tl-wr841hp-v3-router

Signed-off-by: Andy Lee <ykienyco@gmail.com>

andyteq committed:
ath79: add support for TP-Link TL-WR841HP v3

Specifications:
- QCA9533 SoC, 8 MiB nor flash, 64 MiB DDR2 RAM
- 4x Ethernet LAN 10/100, 1x Ethernet WAN 10/100
- 1x LAN, WAN, Wifi, PWR, WPS, RE Leds
- Reset, Wifi on/off, WPS, RE buttons
- Serial UART at J4 onboard: 3.3v GND RX TX, 1152008N1

Installation:

Upload openwrt-ath79-generic-tplink_tl-wr841hp-v3-squashfs-factory.bin
from stock firmware webgui

Revert back to stock firmware instructions:

- set your PC to static IP address 192.168.0.66 netmask 255.255.255.0
- download stock firmware from Tp-link website
- put it in the root directory of tftp server
- rename it to wr841hpv3_tp_recovery.bin
- power on while pressing reset button until any LED is lighting up
- wait for the router to reboot. done

This port has been tested by many people, see here
https://forum.openwrt.org/t/support-for-tp-link-tl-wr841hp-v3-router

Signed-off-by: Andy Lee <ykienyco@gmail.com>

adrianschmutzler commented on target/linux/ath79/dts/qca9533_tplink_tl-wr841hp-v3.dts:
```diff
@@ -0,0 +1,136 @@
+		led-upgrade = &led_pwr;
+	};
+
+	leds: leds {
```
The "leds:" DT label can be dropped here.

andyteq replied:
Did you mean like this ? :
leds: {
	compatible = "gpio-leds";
	wifi {

adrianschmutzler replied:
Just

```
leds {
	...
```

andyteq replied:
ok, I understand, tks.

adrianschmutzler commented on target/linux/ath79/dts/qca9533_tplink_tl-wr841hp-v3.dts:
```diff
@@ -0,0 +1,136 @@
+	leds: leds {
+		compatible = "gpio-leds";
+
+		wifi {
```
Please either call this "wlan" as in the label or "wifi" as in the node name consistently.

adrianschmutzler commented on target/linux/ath79/dts/qca9533_tplink_tl-wr841hp-v3.dts:
```diff
@@ -0,0 +1,136 @@
+		};
+
+		wan_red {
+			label = "tp-link:red:wan";
```
Manual says "orange", unless this is mixed by red and green ...
Note that if you change it, "amber" is actually preferred to orange then.

adrianschmutzler commented on target/linux/ath79/dts/qca9533_tplink_tl-wr841hp-v3.dts:
```diff
@@ -0,0 +1,136 @@
+		wps {
+			label = "tp-link:green:wps";
+			gpios = <&gpio 4 GPIO_ACTIVE_LOW>;
+		};
```
What about RE LED?

adrianschmutzler commented on target/linux/ath79/dts/qca9533_tplink_tl-wr841hp-v3.dts:
```diff
@@ -0,0 +1,136 @@
+			linux,code = <KEY_WPS_BUTTON>;
+			gpios = <&gpio 2 GPIO_ACTIVE_LOW>;
+			debounce-interval = <60>;
+		};
```
What about RE button?

adrianschmutzler commented on target/linux/ath79/dts/qca9533_tplink_tl-wr841hp-v3.dts:
```diff
@@ -0,0 +1,136 @@
+		};
+
+		rfkill {
+			label = "RFKILL button";
```
Consider calling this WiFi button as in the manual.

adrianschmutzler commented on target/linux/ath79/dts/qca9533_tplink_tl-wr841hp-v3.dts:
```diff
@@ -0,0 +1,136 @@
+
+&spi {
+	status = "okay";
+	num-cs = <1>;
```
please add an empty line before and after num-cs

adrianschmutzler commented on target/linux/ath79/dts/qca9533_tplink_tl-wr841hp-v3.dts:
```diff
@@ -0,0 +1,136 @@
+
+&eth0 {
+	status = "okay";
+	phy-handle = <&swphy4>;
```
I'd prefer an empty line after and before phy-handle.

adrianschmutzler commented on target/linux/ath79/dts/qca9533_tplink_tl-wr841hp-v3.dts:
```diff
@@ -0,0 +1,136 @@
+	status = "okay";
+	phy-handle = <&swphy4>;
+	mtd-mac-address = <&uboot 0x1fc00>;
+	mtd-mac-address-increment = <1>;
```
Is the MAC address assignment chosen in accordance to vendor firmware? If yes, I'd be happy about a short note/overview in the commit message (effectively showing the increment of WAN/LAN/Wifi with respect to label MAC address and the fact that you checked it against vendor settings)

andyteq replied:
Ok, I will flash back to vendor firmware & check again.

adrianschmutzler replied:
If you checked, you don't need to check _again_. But if you didn't check, you should check.

andyteq replied:
I'm checked my router u-boot offset at 0x1fc00, it's contain exactly label mac-address.
On vendor, label mac-addres=lan/wifi mac-address. wan mac-address=lan mac-address +1
example my router: lan/wifi mac=D4:6E:0E:XX:XX:EA ; wan mac =D4:6E:0E:XX:XX:EB
The same for openwrt firmware.

adrianschmutzler replied:
Okay, thanks. Then please add these findings to the commit message as well (you might just put this in a short table or something):

```
MAC addresses based on vendor firmware:
LAN    *:ea    label
WAN    *:eb   label +1
2.4 GHz  *:ea   label

The label MAC address in found in u-boot 0x1fc00.
```

andyteq replied:
I committed before you said, how to fix it?

adrianschmutzler replied:
You will have to squash all your changes into one proper commit anyway (use git rebase -i ...). Then, force-push them here again (git push -f ...). Despite, please make formatting nicer than in my quickly typed example.

adrianschmutzler commented on target/linux/ath79/dts/qca9533_tplink_tl-wr841hp-v3.dts:
```diff
@@ -0,0 +1,136 @@
+};
+
+&wmac {
+	status = "okay";
```
add empty line after status.

adrianschmutzler commented on target/linux/ath79/generic/base-files/etc/board.d/01_leds:
```diff
@@ -296,6 +296,11 @@ tplink,tl-mr6400-v1)
 	;;
+tplink,tl-wr841hp-v3)
+	ucidef_set_netdev "wan" "WAN Data" "tp-link:green:wan" "eth1" "tx rx"
+	ucidef_set_netdev "wan_red" "WAN Link" "tp-link:red:wan" "eth1" "link"
```
That seems different from what the manual says; in accordance with most other TP-Link devices, I'd just use green wan LED here.

adrianschmutzler replied:
If you do that, you should be able to just add this device to the case with tplink,archer-c58-v1.

adrianschmutzler commented on target/linux/ath79/generic/base-files/etc/board.d/02_network:
```diff
@@ -292,6 +292,11 @@ ath79_setup_interfaces()
+	tplink,tl-wr841hp-v3)
+		ucidef_set_interface_wan "eth1"
+		ucidef_add_switch "switch0" \
+			"0@eth0" "1:lan:1" "2:lan:2" "3:lan:3" "4:lan:4"
```
This can be add to the case of comfast,cf-e560ac above.

adrianschmutzler commented on target/linux/ath79/image/generic-tp-link.mk:
```diff
@@ -613,6 +613,16 @@ define Device/tplink_tl-wr810n-v2
+  DEVICE_MODEL := TL-WR841HP
+  DEVICE_VARIANT := v3
+  TPLINK_HWID := 0x08411003
+  SUPPORTED_DEVICES += tl-wr841hp-v3
```
This line can be dropped.

andyteq replied:
It will look like this ?
```
define Device/tplink_tl-wr841hp-v3
  $(Device/tplink-8mlzma)
  SOC := qca9533
endef
TARGET_DEVICES += tplink_tl-wr841hp-v3

```

adrianschmutzler replied:
Just drop the SUPPORTED_DEVICES _line_.

andyteq committed:
ath79: add support for TP-Link TL-WR841HP v3

Specifications:
- QCA9533 SoC, 8 MiB nor flash, 64 MiB DDR2 RAM
- 4x Ethernet LAN 10/100, 1x Ethernet WAN 10/100
- 1x WAN, LAN, Wifi, PWR, WPS, RE Leds
- Reset, Wifi on/off, WPS, RE buttons
- Serial UART at J4 onboard: 3.3v GND RX TX, 1152008N1

MAC addresses based on vendor firmware:
LAN    *:ea    label
WAN    *:eb   label +1
2.4 GHz  *:ea   label

The label MAC address in found in u-boot 0x1fc00

Installation:
Upload openwrt-ath79-generic-tplink_tl-wr841hp-v3-squashfs-factory.bin
from stock firmware webgui

Revert back to stock firmware instructions:
- set your PC to static IP address 192.168.0.66 netmask 255.255.255.0
- download stock firmware from Tp-link website
- put it in the root directory of tftp server
- rename it to wr841hpv3_tp_recovery.bin
- power on while pressing reset button until any LED is lighting up
- wait for the router to reboot. done

This support has been tested by many people, see here
https://forum.openwrt.org/t/support-for-tp-link-tl-wr841hp-v3-router

Signed-off-by: Andy Lee <ykienyco@gmail.com>

andyteq committed:
ath79: add support for TP-Link TL-WR841HP v3

Specifications:
- QCA9533 SoC, 8 MiB nor flash, 64 MiB DDR2 RAM
- 4x Ethernet LAN 10/100, 1x Ethernet WAN 10/100
- 1x WAN, LAN, Wifi, PWR, WPS, RE Leds
- Reset, Wifi on/off, WPS, RE buttons
- Serial UART at J4 onboard: 3.3v GND RX TX, 1152008N1

- On the vendor firmware, Label MAC address = LAN/WiFi MAC address
  WAN MAC address = LAN MAC address +1
- The same for Openwrt firmware

Installation:
Upload openwrt-ath79-generic-tplink_tl-wr841hp-v3-squashfs-factory.bin
from stock firmware webgui

Revert back to stock firmware instructions:
- set your PC to static IP address 192.168.0.66 netmask 255.255.255.0
- download stock firmware from Tp-link website
- put it in the root directory of tftp server
- rename it to wr841hpv3_tp_recovery.bin
- power on while pressing reset button until any LED is lighting up
- wait for the router to reboot. done

This support has been tested by many people, see here
https://forum.openwrt.org/t/support-for-tp-link-tl-wr841hp-v3-router

Signed-off-by: Andy Lee <ykienyco@gmail.com>

andyteq committed:
Merge branch 'master' of https://github.com/andyteq/openwrt-1 into myport
